src/deterministic_secure_broadcast.rs

```diff
@@ -193,7 +193,25 @@ impl<A: SecureBroadcastAlgorithm> SecureBroadcastProc<A> {
                     println!("[DSB] we have quorum over msg, sending proof to network");
                     // We have quorum, broadcast proof of agreement to network
                     let proof = self.pending_proof[&msg].clone();
-                    self.broadcast(Payload::ProofOfAgreement { msg, proof })
+                    let add_self = match msg.op {
+                        BFTOp::MembershipNewPeer(id) => {
+                            if id == self.actor() {
+                                true
+                            } else {
+                                println!("[DSB/BFT] Found unexepected non-self peer {}", id);
+                                false
+                            }
+                        },
+                        _ => false
+                    };
```

Comment:
Move this after the broadcast so that we don't have to keep an `add_self` variable (just insert the peer where we have `true`. Also use a match case guard ie.
```rust
match msg.op {
    BFTOp::MembershipNewPeer(id) if id == self.actor() => self.peers.replace(self.actor()))
    _ => ()
};
```

Reply:
OK, I see now why you need the `add_self` variable. I'll keep it as is :+1: